drivers/soundwire/bus.h

```diff
@@ -212,5 +212,6 @@ void sdw_clear_slave_status(struct sdw_bus *bus, u32 request);
 int sdw_slave_modalias(const struct sdw_slave *slave, char *buf, size_t size);
 void sdw_compute_slave_ports(struct sdw_master_runtime *m_rt,
 			     struct sdw_transport_data *t_data);
+int sdw_slave_set_frequency(struct sdw_slave *slave);
```

Comment:
sorry, no this is not the right direction @bardliao 

It is intentional that this initialization is done on startup. The base clock is programmed once, and then we use initial default clock scale registers.

If and when we implement support for the clock change, the only thing that will be updated will be the clock scale registers. The base will *NEVER* be updated because this breaks audio functionality.

So no I don't think this helper should be used when starting a stream. It's not what it was designed for.